cli/src/helpers.rs

```diff
@@ -606,3 +614,88 @@ pub fn command_deposit(
     };
     Ok(result)
 }
+
+#[derive(Serialize)]
+pub struct WithdrawOutput {
+    #[serde(serialize_with = "serialize_b58")]
+    pub from_token_address: Pubkey,
+
+    /// Amount of SOL that was withdrawn.
+    pub withdrawn_sol: Lamports,
+
+    /// Newly created stake account, where the source stake account will be
+    /// split to.
+    #[serde(serialize_with = "serialize_b58")]
+    pub new_stake_account: Pubkey,
+}
+
+impl fmt::Display for WithdrawOutput {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "Withdrawn from:          {}", self.from_token_address)?;
+        writeln!(f, "Total SOL withdrawn:     {}", self.withdrawn_sol)?;
+        writeln!(f, "New stake account:       {}", self.new_stake_account)?;
+        Ok(())
+    }
+}
+
+pub fn command_withdraw(
+    config: &mut SnapshotClientConfig,
+    opts: &WithdrawOpts,
+) -> std::result::Result<WithdrawOutput, crate::error::Error> {
+    let (st_sol_address, new_stake_account) = config.with_snapshot(|config| {
+        let solido = config.client.get_solido(opts.solido_address())?;
+
+        let st_sol_address = spl_associated_token_account::get_associated_token_address(
+            &config.signer.pubkey(),
+            &solido.st_sol_mint,
+        );
+
+        let stake_authority =
+            solido.get_stake_authority(opts.solido_program_id(), opts.solido_address())?;
+
+        // Get heaviest validator.
+        let heaviest_validator = get_validator_to_withdraw(&solido.validators).map_err(|_| {
+            MaintenanceError::new(
```

Comment:
It’s not really a maintenance error. I added a catch-all `CliError` in #343 that we can use for this. If you want to merge first you can also add a
```rs
// TODO(ruuda)
```
comment here and then I’ll change it as part of #343.